### PR TITLE
evidence: remove pubkey from duplicate vote evidence

### PR DESCRIPTION
**Pull request by marbar3778**
this pr brings over the removal of pubkey from duplicatevote evidence from proto-breakage

ref #4580

### CODE REVIEW

File: types/evidence.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 
 const (
 	// MaxEvidenceBytes is a maximum size of any evidence (including amino overhead).
-	MaxEvidenceBytes int64 = 484
+	MaxEvidenceBytes int64 = 444
```
Comment (cmwaters) [inline types/evidence.go:22]: Why did you change this?

Comment (marbar3778) [reply types/evidence.go:22]: the size of duplicatevoteEvidence shrunk. 

I should add table driven tests for all types of evidence for max evidenceBytes

Comment (cmwaters) [reply types/evidence.go:22]: Does the DuplicateVoteEvidence dictate the evidence size? Seems unfair on the other types of evidence

Comment (marbar3778) [reply types/evidence.go:22]: all the new types exceed the max size

```
   TestMaxEvidenceBytes: evidence_test.go:147: 
        	Error Trace:	evidence_test.go:147
        	Error:      	Not equal: 
        	            	expected: int64(444)
        	            	actual  : int(640)
        	Test:       	TestMaxEvidenceBytes
        	Messages:   	LunaticValidatorEvidence
    TestMaxEvidenceBytes: evidence_test.go:147: 
        	Error Trace:	evidence_test.go:147
        	Error:      	Not equal: 
        	            	expected: int64(444)
        	            	actual  : int(649)
        	Test:       	TestMaxEvidenceBytes
        	Messages:   	PhantomValidatorEvidence
    TestMaxEvidenceBytes: evidence_test.go:147: 
        	Error Trace:	evidence_test.go:147
        	Error:      	Not equal: 
        	            	expected: int64(444)
        	            	actual  : int(3124)
        	Test:       	TestMaxEvidenceBytes
        	Messages:   	ConflictingHeadersEvidence
```

I'm not sure how much to bump it as 3124 seems like quite a bit

Comment (cmwaters) [reply types/evidence.go:22]: The conflicting headers doesn't get submitted to the block so I wonder if we count it or not

Comment (cmwaters) [reply types/evidence.go:22]: Just leave it for now and maybe we should look into it with a separate issue - I'm not sure how important the number is

Comment (marbar3778) [reply types/evidence.go:22]: commented it out for now

File: types/evidence_test.go [hunk 2]
```diff
@@ -108,15 +107,47 @@ func TestMaxEvidenceBytes(t *testing.T) {
...
-		PubKey: secp256k1.GenPrivKey().PubKey(), // use secp because it's pubkey is longer
-		VoteA:  makeVote(t, val, chainID, math.MaxInt64, math.MaxInt64, math.MaxInt64, math.MaxInt64, blockID),
-		VoteB:  makeVote(t, val, chainID, math.MaxInt64, math.MaxInt64, math.MaxInt64, math.MaxInt64, blockID2),
+		VoteA: makeVote(t, val, chainID, math.MaxInt64, math.MaxInt64, math.MaxInt64, math.MaxInt64, blockID),
+		VoteB: makeVote(t, val, chainID, math.MaxInt64, math.MaxInt64, math.MaxInt64, math.MaxInt64, blockID2),
 	}
 
-	bz, err := cdc.MarshalBinaryLengthPrefixed(ev)
-	require.NoError(t, err)
+	// evl := &LunaticValidatorEvidence{
```
Comment (melekes) [inline types/evidence_test.go:114]: is this a TODO?

Comment (marbar3778) [reply types/evidence_test.go:114]: Left it commented because I'm not sure if we should bump the max bytes higher or leave as is.
